gpflow/models/sgpr.py

```diff
@@ -59,10 +59,12 @@ def upper_bound(self):
         kuu = Kuu(self.inducing_variable, self.kernel, jitter=default_jitter())
         kuf = Kuf(self.inducing_variable, self.kernel, x_data)
 
-        KufKfu = tf.linalg.matmul(kuf, kuf, transpose_b=True)
+        I = tf.eye(tf.shape(kuu)[0], dtype=default_float())
 
         L = tf.linalg.cholesky(kuu)
-        LB = tf.linalg.cholesky(kuu + KufKfu / self.likelihood.variance)
+        A = tf.linalg.triangular_solve(L, kuf, lower=True)
+        B = I + tf.linalg.matmul(A, A, transpose_b=True) / self.likelihood.variance
```

Comment:
You can compute it only once:  tf.linalg.matmul(A, A, transpose_b=True)

Reply:
Will do.